### PR TITLE
Rename astro namespace to celestia::astro

### PR DESCRIPTION
**Pull request by ajtribick**
- Tidy up includes in astro.h/astro.cpp
- Replace snprintf with fmt in astro.cpp

### CODE REVIEW

File: src/celengine/astro.cpp [hunk 1]
```diff
@@ -321,15 +373,19 @@ astro::Date::Date(double jd)
     tzname = "UTC";
 }
 
-const char* astro::Date::toCStr(Format format) const
+
+const char*
+Date::toCStr(Format format) const
 {
-    static char date[255];
+    static std::array<char, 255> date;
```
Comment (375gnu) [inline src/celengine/astro.cpp:380]: if you're in mood could you make this thread safe? I hope one day we can start using more then 1 core.

File: src/celengine/boundaries.cpp [hunk 2]
```diff
@@ -74,8 +74,8 @@ ReadBoundaries(std::istream& in)
             currentChain.clear();
         }
 
-        currentChain.emplace_back(astro::equatorialToCelestialCart(ra, dec,
-                                                                   BoundariesDrawDistance));
+        currentChain.emplace_back(celestia::astro::equatorialToCelestialCart(ra, dec,
```
Comment (375gnu) [inline src/celengine/boundaries.cpp:77]: better consistently use `namespace astro = celestia::astro;`, then one day when all code will be in `celestia` namespace no changes will be required.

File: src/celengine/astro.cpp (hunk 1 above)
Comment (ajtribick) [reply src/celengine/astro.cpp:380]: Yep, makes sense. Usages don't seem to be on any particularly critical paths, so might as well just return a `std::string`

File: src/celengine/astro.cpp [hunk 3]
```diff
@@ -364,35 +419,35 @@ const char* astro::Date::toCStr(Format format) const
         break;
     }
 
-    strftime(date, sizeof(date), strftime_format, &cal_time);
+    auto length = std::strftime(result.data(), result.size(), strftime_format, &cal_time);
```
Comment (375gnu) [inline src/celengine/astro.cpp:422]: could you check if we can use `fmt` here as well?

Comment (ajtribick) [reply src/celengine/astro.cpp]: Have put something in but am currently on Windows, so need to do some testing with this code path active.

**Comment by ajtribick** [comment]
I'm not prepared to mess with the internals of the JD/calendar conversions right now, hence sprinkling `//NOSONAR` all over it.

File: test/unit/kepler_test.cpp [hunk 4]
```diff
@@ -9,6 +9,9 @@
 
 #include <doctest.h>
 
+namespace astro = celestia::astro;
+
+
```
Comment (375gnu) [inline test/unit/kepler_test.cpp:14]: let's use only empty line as a separator

**Review by 375gnu — CHANGES_REQUESTED** [review]

**Review by 375gnu — APPROVED** [review]

File: src/celengine/astro.cpp [hunk 5]
```diff
@@ -321,15 +401,14 @@ astro::Date::Date(double jd)
...
 
+std::string
+Date::toString(Format format) const
+{
     if (format == ISO8601)
     {
-        snprintf(date, sizeof(date), "%04d-%02d-%02dT%02d:%02d:%08.5fZ",
-                 year, month, day, hour, minute, seconds);
-        return date;
+        return fmt::format("{:04}-{:02}-{:02}T{:02}:{:02}:{:02}:{:08.5f}Z"sv,
```
Comment (levinli303) [inline src/celengine/astro.cpp:410]: @ajtribick we missed an argument here

**Pull request merged**